ci/gh-experiment: sharness fails: Downloaded SaxonHE11-4J.zip has wrong md5sum

"Run Sharness tests":

```
*** test/sharness/download-saxon ***
Downloading SaxonHE11-4J.zip
Downloaded SaxonHE11-4J.zip has wrong md5sum ('291fc26f7a9fcf73b9425e27592e6051' != '8a4783d307c32c898f8995b8f337fd6b')
[...]
Error: Process completed with exit code 2.
```

Log: https://github.com/ipfs/kubo/actions/runs/3959053284/jobs/6781377209#step:17:50

@galargh perhaps we could add these artifacts to IPFS and fetch them from gateway like we [do with webui in ipfs-desktop](https://github.com/ipfs/ipfs-desktop/blob/8aca5c268aea9d7066f2853639239563e4606545/package.json#L20) + add 2-3x  retry so we don't break CI on HTTP hiccups?

--- a/test/sharness/lib/download-saxon.sh
+++ b/test/sharness/lib/download-saxon.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 dependencies=(
-  "url=https://sourceforge.net/projects/saxon/files/Saxon-HE/11/Java/SaxonHE11-4J.zip;md5=8a4783d307c32c898f8995b8f337fd6b"
+  "url=https://raw.githubusercontent.com/pl-strflt/Saxon-HE/3e039cdbccf4efb9643736f34c839a3bae3402ae/11/Java/SaxonHE11-4J.zip;md5=8a4783d307c32c898f8995b8f337fd6b"
   "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-frames-saxon.xsl;md5=6eb013566903a91e4959413f6ff144d0"
   "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-noframes-saxon.xsl;md5=8d54882d5f9d32a7743ec675cc2e30ac"
 )

